### PR TITLE
refactor: use `exactly one` throughout code base

### PR DESCRIPTION
**Pull request by natygyoon**
exactly_one is a neat pattern introduced in [this PR](https://github.com/Unstructured-IO/unstructured/pull/343) that could be leveraged in quite a few other places in the code base. For example, it could be added [here](https://github.com/Unstructured-IO/unstructured/blob/a9152313aae30f7a7a6cf8bddd512afd9d65c3ec/unstructured/file_utils/filetype.py#L131).

### CODE REVIEW

**Comment by ryannikolaidis** [comment]
nit: it looks like this branches from your --metadata-include/exclude branch. because this includes all of those changes, it makes it difficult to review these changes specifically.

File: test_unstructured_ingest/test-ingest-github.sh [hunk 1]
```diff
@@ -4,9 +4,9 @@ SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 cd "$SCRIPT_DIR"/.. || exit 1
 
 if [[ "$CI" == "true" ]]; then
-    if [ "$(( RANDOM % 10))" -lt 2 ] ; then
+    if [ "$(( RANDOM % 10))" -lt 1 ] ; then
         # NOTE(crag): proper fix is being tracked here: https://github.com/Unstructured-IO/unstructured/issues/306
-        echo "Skipping ingest 80% of github ingest tests to avoid rate limiting issue."
+        echo "Skipping ingest 90% of github ingest tests to avoid rate limiting issue."
```
Comment (ryannikolaidis) [inline test_unstructured_ingest/test-ingest-github.sh:9]: what was this about? keeping the time down?

File: unstructured/ingest/main.py [hunk 2]
```diff
@@ -372,7 +378,14 @@ def main(
...
     metadata_exclude,
+    fields_include,
 ):
+    if "metadata" not in fields_include and (metadata_include or metadata_exclude):
+        logger.error(
+            "Either `--metadata-include` or `--metadata-exclude` is specified"
+            " while metadata is not specified in --fields-include.",
+        )
+        sys.exit(1)
     if not preserve_downloads and download_dir:
```
Comment (ryannikolaidis) [inline unstructured/ingest/main.py:389]: good catch

**Review by ryannikolaidis — CHANGES_REQUESTED** [review]
The changes that pertain to this PR all look good! I can't approve because you have changes from other PRs mixed in. If you want to wait until those merge and then rebase this or make these change against main, then I can approve.

File: unstructured/ingest/main.py [hunk 3]
```diff
@@ -35,6 +36,35 @@
...
+                    ", ".join(self.mutually_exclusive),
+                ),
+            )
+
+        return super(MutuallyExclusiveOption, self).handle_parse_result(
+            ctx,
+            opts,
+            args,
+        )
+
```
Comment (ryannikolaidis) [inline unstructured/ingest/main.py:66]: is this meant to be in this PR?

Comment (natygyoon) [reply unstructured/ingest/main.py:66]: Nope, I must have made a mistake while merging main in.

**Review by cragwolfe — APPROVED** [review]
LGTM!

**Review by ryannikolaidis — APPROVED** [review]
Lgtm!

**Pull request merged**